spec/unit/cloudkeeper/aws/cloud_spec.rb

```diff
@@ -0,0 +1,161 @@
+                  n = f[:name] == t[:key]
+                  v = f[:values].empty? ? true : f[:values].include?(t[:value])
+                  n && v
+                end
+              end
+
+              { tags: filtered_tags }
+            end)
+          end
+
+          it 'returns empty array' do
+            tags = cloud.search_tags([{ name: 'cloudkeeper', values: [] }])
+            expect(tags).to be_empty
+          end
+        end
+
+        context 'with no image resources' do
+          before do
+            stub_tags = [
+              { key: 'cloudkeeper', resource_id: '0', resource_type: 'instance', value: 'yes' },
+              { key: 'image-list-id', resource_id: '0', resource_type: 'instance', value: '8' }
+            ]
+
+            ec2.stub_responses(:describe_tags, lambda do |context|
+              filtered_tags = stub_tags.select do |t|
+                context.params[:filters].find do |f|
+                  n = f[:name] == t[:key]
+                  v = f[:values].empty? ? true : f[:values].include?(t[:value])
+                  n && v
+                end
+              end
+
+              { tags: filtered_tags }
+            end)
+          end
+
+          it 'searches for cloudkeeper tag' do
+            tags = cloud.search_tags([{ name: 'cloudkeeper', values: [] }])
+            expect(tags).to be_empty
+          end
+
+          it 'searches for image-list-id tag and value' do
+            tags = cloud.search_tags([{ name: 'image-list-id', values: ['8'] }])
+            expect(tags).to be_empty
+          end
+        end
+
+        context 'with multiple resources' do
+          before do
+            stub_tags = [
+              { key: 'cloudkeeper', resource_id: '0', resource_type: 'instance', value: 'yes' },
+              { key: 'image-list-id', resource_id: '0', resource_type: 'instance', value: '8' },
+              { key: 'cloudkeeper', resource_id: '1', resource_type: 'image', value: 'yes' },
+              { key: 'image-list-id', resource_id: '1', resource_type: 'image', value: '8' }
+            ]
+
+            ec2.stub_responses(:describe_tags, lambda do |context|
+              filtered_tags = stub_tags.select do |t|
+                context.params[:filters].find do |f|
+                  n = f[:name] == t[:key]
+                  v = f[:values].empty? ? true : f[:values].include?(t[:value])
+                  n && v
+                end
+              end
+
+              { tags: filtered_tags }
+            end)
+          end
+
+          it 'searches for cloudkeeper tag' do
+            tags = cloud.search_tags([{ name: 'cloudkeeper', values: [] }])
+            expect(tags.length).to eq(1)
+          end
+
+          it 'searches for image-list-id tag and value' do
+            tags = cloud.search_tags([{ name: 'image-list-id', values: ['8'] }])
+            expect(tags.length).to eq(1)
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
What about tests for the rest of the methods?

Reply:
Me not that kind of orc!

Reply:
I dont belive that it is neccesary to test functionality that is purely AWS library. But feel free to change my mind.